source/fundamentals/connection.txt

```diff
@@ -0,0 +1,435 @@
+================
+Connection Guide
+================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. toctree::
+
+   /fundamentals/connection/tls
+   /fundamentals/connection/jndi
+
+This guide shows you how to connect to a MongoDB instance or replica set
+deployment using the driver.
+
+For information about authenticating with a MongoDB instance,
+see :doc:`Authentication </fundamentals/auth>`. To learn more about using
+the driver with the Java Naming and Directory Interface (JNDI), see
+:doc:`JNDI <fundamentals/connection/jndi>`. To configure TLS/SSL security
+for connections to your MongoDB instance, see :doc:`TLS/SSL
+<fundamentals/connection/tls>`.
+
+.. _mongoclient:
+
+MongoClient
+-----------
+
+Use the ``MongoClients.create()`` method to construct a
+``MongoClient``. As each ``MongoClient`` represents a threadsafe pool of
+connections to the database, most applications only require a single
+instance of a ``MongoClient``, even across multiple threads. All resource
+usage limits, such as max connections, apply to individual
+``MongoClient`` instances.
+
+.. note::
+
+   Always call ``MongoClient.close()`` to clean up resources when an
+   instance is no longer needed. The `try-with-resources statement
```

Comment:
`try-with-resources` is not that useful in practice, as it requires that the lifetime of the MongoClient is scoped to a single method.  In a typical web application, where you want to cache the MongoClient for the life of that application, this is not generally the case.  In fact, if you're doing that, you're usually doing something wrong, because you are creating a MongoClient per web request!

Reply:
I suggest removing this.

Reply:
Seems fair -- try-with-resources makes sense for usage examples, doesn't make sense for enterprise applications.